src/pudl/transform/ferc714.py

```diff
@@ -0,0 +1,483 @@
+        fall = time_changes.loc[yr, "fall_back"]
+        std_times = (
+            (df.respondent_id == respondent_id) &
+            (df.local_time.dt.year == yr) &
+            ((df.local_time < spring) | (df.local_time >= fall))
+        )
+        dst_times = (
+            (df.respondent_id == respondent_id) &
+            (df.local_time.dt.year == yr) &
+            ((df.local_time >= spring) & (df.local_time < fall))
+        )
+        df.loc[std_times, tz_col] = std_val
+        df.loc[dst_times, tz_col] = dst_val
+    return df
+
+
+def _apply_tz_fixes_by_date(df, fixes, time_changes):
+    """A wrapper assin many timezone fixes at once."""
+    for fix in fixes:
+        df = _fix_tz_by_date(df, time_changes=time_changes, **fix)
+    return df
+
+
+def respondents_ferc714(pudl_settings):
+    """Table of FERC 714 respondents IDs, names, and EIA utility IDs."""
+    zippath = zipfile.Path(
+        pathlib.Path(pudl_settings["data_dir"], "ferc714/data/ferc714.zip"),
+        "Respondent IDs.csv"
+    )
+    with zippath.open() as zp:
+        df = pd.read_csv(zp)
+    df = (
+        df.astype({
+            "eia_code": pd.Int64Dtype(),
+            "respondent_id": pd.Int64Dtype(),
+            "respondent_name": pd.StringDtype(),
+        })
+        .assign(
+            respondent_name=lambda x: x.respondent_name.str.strip(),
+            eia_code=lambda x: x.eia_code.replace(to_replace=0, value=pd.NA)
+        )
+        # These excludes fake Test IDs -- not real planning areas
+        .query("respondent_id not in @bad_respondents")
+    )
+    return df
+
+
+def planning_area_hourly_demand_ferc714(pudl_settings):
+    """Hourly demand time series by Planning Area."""
+    zippath = zipfile.Path(
+        pathlib.Path(pudl_settings["data_dir"], "ferc714/data/ferc714.zip"),
+        "Part 3 Schedule 2 - Planning Area Hourly Demand.csv"
+    )
+    with zippath.open() as zp:
+        df = pd.read_csv(zp)
+
+    df = (
+        df.filter(regex=r"^(?!.*_f$).*")
+        .drop(["report_yr", "report_prd", "spplmnt_num", "row_num"], axis="columns")
+        # Fake test respondents
+        .query("respondent_id not in @bad_respondents")
+        .assign(
+            plan_date=lambda x: pd.to_datetime(x.plan_date),
+            timezone=lambda x: x.timezone.str.upper().str.strip(),
+        )
+        .rename(columns=_hours_to_ints)
+        .melt(
+            id_vars=["respondent_id", "plan_date", "timezone"],
+            var_name="hour",
+            value_name="demand_mwh"
+        )
+        .pipe(_fix_tz_ferc714, tz_map=tz_nulls, tz_fixes=tz_fixes)
+        .pipe(_fix_dst, spring_forward, fall_back)
+        .pipe(_apply_tz_fixes_by_date, tz_fixes_by_date, time_changes)
+        .merge(
+            respondents_ferc714(pudl_settings),
+            how="left",
+            on="respondent_id"
+        )
+        .sort_values(["respondent_id", "local_time"])
+    )
+    return df
+
+
+def electricity_planning_areas(pudl_settings):
```

Comment:
my only comment in here is that it might be nice to immediately clean the column names with `helpers.cleanstrings_snake` to fit with our conventions. I assume you'll want to do that eventually and it would probably be better to do it before you need to use the column names a ton.

Reply:
I kind of held off on trying to munge the geospatial data since I'm not really sure how we're going to end up saving them and sharing the info. For now I just wanted to get it read into a dataframe so I could see whether it could be joined to the demand data from 714.  But yes, clearly these aren't the names we'd really use.